### ISSUE TITLE
Network failure installing puppet modules

### ISSUE THREAD

**Issue by johnsca** [juju-solutions/layer-apache-bigtop-base#54]
```
2016-11-15 00:02:20 INFO install [mNotice: Preparing to install into /etc/puppet/modules ...[0m
2016-11-15 00:02:20 INFO install [mNotice: Module puppetlabs-stdlib 4.13.1 is already installed.[0m
2016-11-15 00:02:21 INFO install [mNotice: Preparing to install into /etc/puppet/modules ...[0m
2016-11-15 00:02:21 INFO install [mNotice: Downloading from https://forgeapi.puppetlabs.com ...[0m
2016-11-15 00:02:44 INFO install [1;31mError: Request to Puppet Forge failed.
2016-11-15 00:02:44 INFO install   The server being queried was https://forgeapi.puppetlabs.com/v3/releases?module=puppetlabs-apt
2016-11-15 00:02:44 INFO install   The HTTP response we received was '502 Bad Gateway'[0m
2016-11-15 00:02:44 INFO install Traceback (most recent call last):
2016-11-15 00:02:44 INFO install   File "/var/lib/juju/agents/unit-slave-1/charm/hooks/install", line 19, in <module>
2016-11-15 00:02:44 INFO install     main()
2016-11-15 00:02:44 INFO install   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/__init__.py", line 78, in main
2016-11-15 00:02:44 INFO install     bus.dispatch()
2016-11-15 00:02:44 INFO install   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 434, in dispatch
2016-11-15 00:02:44 INFO install     _invoke(other_handlers)
2016-11-15 00:02:44 INFO install   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 417, in _invoke
2016-11-15 00:02:44 INFO install     handler.invoke()
2016-11-15 00:02:44 INFO install   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 291, in invoke
2016-11-15 00:02:44 INFO install     self._action(*args)
2016-11-15 00:02:44 INFO install   File "/var/lib/juju/agents/unit-slave-1/charm/reactive/apache_bigtop_base.py", line 27, in fetch_bigtop
2016-11-15 00:02:44 INFO install     Bigtop().install()
2016-11-15 00:02:44 INFO install   File "lib/charms/layer/apache_bigtop_base.py", line 129, in install
2016-11-15 00:02:44 INFO install     self.install_puppet_modules()
2016-11-15 00:02:44 INFO install   File "lib/charms/layer/apache_bigtop_base.py", line 227, in install_puppet_modules
2016-11-15 00:02:44 INFO install     utils.run_as('root', 'puppet', 'module', 'install', 'puppetlabs-apt')
2016-11-15 00:02:44 INFO install   File "/usr/local/lib/python3.5/dist-packages/jujubigdata/utils.py", line 361, in run_as
2016-11-15 00:02:44 INFO install     return run(['su', user, '-c', quoted], env=env, stdin=stdin)
2016-11-15 00:02:44 INFO install   File "/usr/lib/python3.5/subprocess.py", line 581, in check_call
2016-11-15 00:02:44 INFO install     raise CalledProcessError(retcode, cmd)
2016-11-15 00:02:44 INFO install subprocess.CalledProcessError: Command '['su', 'root', '-c', "'puppet' 'module' 'install' 'puppetlabs-apt'"]' returned non-zero exit status 1
2016-11-15 00:02:44 ERROR juju.worker.uniter.operation runhook.go:107 hook "install" failed: exit status 1
```

This error was transient, but we should handle it.  In furtherance of #42 we should probably just bundle them in and [install from local](https://docs.puppet.com/puppet/latest/reference/modules_installing.html#install-from-a-release-tarball) instead of relying on the network.

**Comment by johnsca** [comment]
Looks like stdlib is already installed, so we can just drop that one.  The tarball release of puppetlabs-apt can be downloaded from [here](https://forge.puppet.com/v3/files/puppetlabs-apt-2.3.0.tar.gz).

**Comment by kwmonroe** [comment]
To clarify (not saying it's right), but does an env with hook retries eventually succeed?  Or are you saying we'll fail puppet installs in places like network-restricted envs?  The latter is quite a different can-o-worms.

**Comment by kwmonroe** [comment]
Fixed in #58 by packing the -stdlib and -apt puppet modules into the base layer and calling `puppet module install` on the local modules.
